homeassistant/components/switch/raspyrfm.py

```diff
@@ -0,0 +1,157 @@
+"""
+Support for switch devices that can be controlled using the RaspyRFM rc module.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch/raspyrfm/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, CONF_PLATFORM, CONF_PORT, CONF_SWITCHES,
+    DEVICE_DEFAULT_NAME)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['raspyrfm-client==1.2.8']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_GATEWAY_MANUFACTURER = 'gateway_manufacturer'
+CONF_GATEWAY_MODEL = 'gateway_model'
+CONF_CONTROLUNIT_MANUFACTURER = 'controlunit_manufacturer'
+CONF_CONTROLUNIT_MODEL = 'controlunit_model'
+CONF_CHANNEL_CONFIG = 'channel_config'
+DEFAULT_HOST = '127.0.0.1'
+
+# define configuration parameters
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): 'raspyrfm',
+    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=None):
+        vol.All(vol.Coerce(int), vol.Range(min=1, max=65535)),
+    vol.Optional(CONF_GATEWAY_MANUFACTURER): cv.string,
+    vol.Optional(CONF_GATEWAY_MODEL): cv.string,
+    vol.Required(CONF_SWITCHES, default=[]): vol.Schema([{
+        vol.Optional(CONF_NAME, default=DEVICE_DEFAULT_NAME): cv.string,
+        vol.Required(CONF_CONTROLUNIT_MANUFACTURER): cv.string,
+        vol.Required(CONF_CONTROLUNIT_MODEL): cv.string,
+        vol.Required(CONF_CHANNEL_CONFIG): cv.match_all,
```

Comment:
Don't allow any value. Everything should be validated as strict as possible.

Reply:
I don't think this is possible as the configuration depends on the controlunit manufacturer and model. All I can say for sure is that every channel config consists of some key value pairs but specific key names do vary. So the best I can do is `cv.Dict`?

Reply:
Do we know the type of the keys in the dict? Eg:

```py
{cv.string: cv.match_all}
```
Try to narrow down the keys and values types of the dict as far as possible.

Reply:
Yes we can do that.